engine/lib/cron.php

```diff
@@ -107,6 +107,13 @@ function _elgg_cron_page_handler($page) {
 		forward();
 	}
 
+	if (get_config('security_protect_cron')) {
+		$token = get_input('ct');
+		if (!_elgg_cron_validate_token($token)) {
```

Comment:
We have added `elgg_http_get_signed_url()`, `elgg_http_validate_signed_url()`, `elgg_signed_request_gatekeeper()` for these use cases.